### PR TITLE
Implement config system partially

### PR DESCRIPTION
**Pull request by ItayNaveh**
without config file and defualts

implemented command line arguments (see TODO)
and enviorment variables

### CODE REVIEW

File: src/snadra/_config/__init__.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    "postgres_user": None,
+}
+
+
+def setupConfig():
+    # cmd args
+
+    # TODO:
+    # maybe a list comprehension to get all keys from `config` and add '=' to the end
+    options, _ = getopt.getopt(sys.argv[1:], "", ["postgres_host=", "postgres_user="])
```
Comment (ShaharNaveh) [inline src/snadra/_config/__init__.py:25]: why not use ```argparse```?

File: src/snadra/_config/__init__.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+import sys
+import typing
+
+config: typing.Dict[str, typing.Optional[str]] = {
+    "postgres_host": None,
+    "postgres_user": None,
+}
+
+
+def setupConfig():
```
Comment (ShaharNaveh) [inline src/snadra/_config/__init__.py:20]:
```suggestion
def setup_config():
```

We are not coding in java:)

File: src/snadra/__main__.py [hunk 3]
```diff
@@ -11,9 +11,12 @@
 """
 import asyncio
 
+from snadra._config import setupConfig
 from snadra._core.parsers import CommandParser
 
 if __name__ == "__main__":
+    setupConfig()
```
Comment (ShaharNaveh) [inline src/snadra/__main__.py:18]:
```suggestion
    setup_config()
```

We are not coding in java:)

**Comment by ItayNaveh** [comment]
should the config path be:
`~/.config/snadra.toml`
or
`~/.config/snadra/config.toml`
?

**Comment by ShaharNaveh** [comment]
> should the config path be:
> `~/.config/snadra.toml`
> or
> `~/.config/snadra/config.toml`
> ?

The later. (```~/.config/snadra/config.toml```)

**Review by ShaharNaveh — CHANGES_REQUESTED** [review]
Overall you did implement features that we will want (one day), but for now this is an overkill.
Let's KISS, and just have one function that reads the config file, and returns a dict:)

But in general, you do have some great ideas, but for now lets KISS

**Comment by ItayNaveh** [comment]
so what do i need to change?

**Comment by ShaharNaveh** [comment]
> so what do i need to change?

Basically everything.
It's not that the code is not good, it's just that the code is not needed (ATM).

Write **one** function, that get the config file path as an argument, and returns a python dictionary of the parsed toml file.

File: src/snadra/_config/__init__.py [hunk 4]
```diff
@@ -1,59 +1,11 @@
...
-    with config_path.open() as f:
-        config_file_data = toml.loads(f.read())
 
-    for config_name in config.keys():
-        if config[config_name] is None:
-            if (env_name := f"SNADRA_{config_name.upper()}") in os.environ:
-                config[config_name] = os.environ[env_name]
+def parse_config_file(path: pathlib.Path) -> typing.Dict[str, typing.Any]:
+    with path.open() as f:
+        config_file_data = rtoml.loads(f.read())
```
Comment (ShaharNaveh) [inline src/snadra/_config/__init__.py:9]:
```suggestion
        config_file_data = rtoml.load(f)
```

**Comment by ShaharNaveh** [comment]
Beside the comment this is a LGTM.

Also once the CI goes green, you can merge:)

**Comment by ItayNaveh** [comment]
@ShaharNaveh fix your ci
it shouldn't complain

**Review by ShaharNaveh — APPROVED** [review]

**Comment by ShaharNaveh** [comment]
Thanks @ItayNaveh

**Pull request merged**